src/main/java/com/mkopec/ticketbookingapp/service/SeatValidationService.java

```diff
@@ -0,0 +1,199 @@
+package com.mkopec.ticketbookingapp.service;
+
+import com.mkopec.ticketbookingapp.domain.RoomSeat;
+import com.mkopec.ticketbookingapp.exception.ArgumentNotValidException;
+import org.springframework.stereotype.Service;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.stream.Collectors.groupingBy;
+
+@Service
+public class SeatValidationService {
+    private enum SeatType {
+        FREE_INSIDE,
+        FREE_ON_EDGE,
+        OCCUPIED,
+        RESERVATION_DONE
+    }
+
+    private static class SeatModel {
+        private SeatType seatType;
+        private RoomSeat roomSeat;
+        private boolean reserved = false;
+
+        private SeatModel(RoomSeat roomSeat, SeatType seatType) {
+            this.seatType = seatType;
+            this.roomSeat = roomSeat;
+        }
+
+        String getRow() {
+            return roomSeat.getSeat().getRow();
+        }
+
+        Integer getNumber() {
+            return roomSeat.getSeat().getNumber();
+        }
+    }
+
+    private static final int NUM_OF_NEIGHBOURS = 2;
+
+    boolean validate(List<RoomSeat> reservedSeats, List<RoomSeat> occupiedSeats, List<RoomSeat> allSeats) throws ArgumentNotValidException {
+        Map<RoomSeat, SeatModel> seats = initializeSeatModels(reservedSeats, occupiedSeats, allSeats);
+
+        // sorting allows to take neighbour of seat
+        Map<String, List<SeatModel>> groupedSeatModelsToRows = seats.values().stream()
+                .sorted(Comparator.comparing(SeatModel::getRow).thenComparing(SeatModel::getNumber))
+                .collect(groupingBy(SeatModel::getRow));
+
+        for (Map.Entry<String, List<SeatModel>> entry : groupedSeatModelsToRows.entrySet()) {
+            List<SeatModel> models = entry.getValue();
+            for (int i = 0; i < models.size(); i++) {
+                SeatModel model = models.get(i);
+                if (model.reserved) {
+                    // check if is possible to reserve
```

Comment:
dlaczego to nie jest nazwa metody, która by to robiła?

Reply:
Racja, to co poniżej powinno być opakowane w taką metodę